### PR TITLE
Hyper convection lathes and industrial ore processor

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Partly #21055 and partly #21462

Replaces the main useful lathe upgrades with 3 new machines. Autolathes and Protolathes now have corresponding hyper-convection varieties. These have a 30% material reduction (between T3 and T4 balancing currently) at the cost of a slightly slower print time and producing heat in the atmosphere while printing (AL upstream???)

The ore processor has had its upgrades replaced with the industrial ore processor, which has a 25% material reduction and a 50% speed increase. This one doesn't have any downsides in order to try and foster that sweet sweet salv/sci communication.

The new lathes are available in the Industrial Engineering tech (repriced to 10k) and the industrial ore processor is in the salvage tech. 

## Why / Balance

Lathe upgrades are some of the most blatantly dominant upgrades in the game. It's not a stretch to say that materials and protolathes/autolathes are science's sole method of actually utilizing the fruits of their research. Higher tiers of machine parts are utilized almost exclusively for their interactions with the lathe, since they are used so commonly and the cost reduction is so effective over time.

By balancing these boosts with technically manageable downsides (heat can be funneled out or nullified by spacing) the upgrades can remain reasonably balanced while additionally being more discoverable and decoupled from the experimental discipline.

The ore reclaimer upgrade has been made pretty trivially available to the benefit of salvage, who can use it to aid the entire station in terms of material supply.

## Media

image
sprites courtesy of devi (ubaserb)

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added the Hyper convection protolathe and autolathe to Industrial Engineering tech. These lathes have a discounted printing price at the cost of venting hot air into the surroundings.
- add: Added the industrial ore processor, an improved version with discounted printing prices, to salvage equipment technology.
- tweak: Moved autolathe, protolathe, circuit imprinter, ore processor, and material reclaimer boards to be available by default in the circuit imprinter.
- remove: Removed machine part upgrades from all lathes.

### CODE REVIEW

File: Resources/Prototypes/Recipes/Lathes/electronics.yml [hunk 1]
```diff
@@ -188,6 +188,22 @@
      Steel: 100
      Glass: 900
 
+- type: latheRecipe
+  id: AutolatheHyperConvectionMachineCircuitboard
+  result: AutolatheHyperConvectionMachineCircuitboard
+  completetime: 4
+  materials:
+    Steel: 100
+    Glass: 900
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Lathes/electronics.yml:197]: should these not require gold like ore processor :trollface:

**Comment by mirrorcult** [comment]
these fuck supa hard

**Comment by EmoGarbage404** [comment]
> these fuck supa hard

yeah i think that heat production strikes a nice middleground where it's manageable with decent setup and makes them less of a "why would you ever NOT get this"

**Comment by Aisu9** [comment]
being manageable if not used 24/7 is interesting, because this will mean antags(or anyone tbh) will be able to fuck over departments if they ever decide to queue up a lot of craftings. all in all foster upgrades are spess and love

**Comment by deltanedas** [comment]
shrimply
1. make freezer
2. make passive vent

File: Content.Server/Lathe/LatheSystem.cs [hunk 2]
```diff
@@ -103,6 +103,31 @@ public override void Update(float frameTime)
...
+
+            var heatQuery = EntityQueryEnumerator<LatheHeatProducingComponent, LatheProducingComponent, TransformComponent>();
+            while (heatQuery.MoveNext(out var uid, out var heatComp, out _, out var xform))
+            {
+                if (_timing.CurTime < heatComp.NextSecond)
+                    continue;
+                heatComp.NextSecond += TimeSpan.FromSeconds(1);
+
+                var position = _transform.GetGridTilePositionOrDefault((uid,xform));
+                var environments = new List<GasMixture>();
```
Comment (metalgearsloth) [inline Content.Server/Lathe/LatheSystem.cs:115]: Just store the list on the system and re-use it.

**Pull request merged**